ut_assert/scripts/generate_stubs.pl

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/perl
+
+#
+#  NASA Docket No. GSC-18,370-1, and identified as "Operating System Abstraction Layer"
+#
+#  Copyright (c) 2019 United States Government as represented by
+#  the Administrator of the National Aeronautics and Space Administration.
+#  All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+#
+# Stub function generator script
+# ------------------------------
+#
+# PURPOSE:
+#   Given a set of public API C header files, this generates a matching set of
+#   stub functions to be used with coverage testing.  Generating the stub functions
+#   in this manner ensures the argument and return types are consistent between
+#   the stub and the declaration.
+#
+# USAGE:
+#   generate_stubs.pl <output-stub-dir> <header-file> ...
+#
+# The script will scan each header file given on the command line, and search them
+# for C function prototypes.  Each prototype will then be used to define a matching stub
+# implementation in the output stub directory.  A .c file will be created with the same
+# basename as the .h file containing stub versions of the functions defined in that header.
+#
+# File boilerplate info (i.e. first block comment) is preserved between input and output file
+# such that this should be able to work with any subsystem without altering the legal info
+# in the file header.
+#
+# By default the stubs are rendered with a default handler following the pattern:
+#    UT_DefaultHandler_${STUBNAME}
+#
+# By default the stubs are written to a file with the same basename and a "-stubs" suffix.
```

Comment:
Minor note: My files came out named + "_stubs" and not "-stubs"

Reply:
actually, that was intentional.  File naming patterns in CFE/OSAL are not consistent, some use `-` and some use `_` as the word separator char.  So the script has a little "when in rome" feature to use whatever char it already sees in the filenames.

Also worth noting (I think the comments say this too) that by default the script will check for the existence of a default handler first, and only render a ref to that default handler if it finds it, otherwise it uses NULL.   This logic was just intended as a way to evolve over time, not forcing one to implement a default handler if it isn't really needed.  (e.g. a common/boring function with no output params and returns `int32` actually doesn't need a handler at all).  So rather than forcing one to implement an empty handler just to fill the gap, the script uses NULL if it doesn't see one implemented.

Reply:
Ha ha, I was expecting it to not overwrite my current file and was surprised when it did.